pkg/noderesourcetopology/numaresources.go

```diff
@@ -52,3 +133,87 @@ func isNUMAAffineResource(resource corev1.ResourceName) bool {
 	// We can't tell for sure, so we default to "no".
 	return false
 }
+
+func isResourceSetSuitable(qos corev1.PodQOSClass, resource corev1.ResourceName, quantity, numaQuantity resource.Quantity) bool {
+	if qos != corev1.PodQOSGuaranteed && isNUMAAffineResource(resource) {
+		return true
+	}
+	return numaQuantity.Cmp(quantity) >= 0
+}
+
+// subtractResourcesFromNUMANodeList finds the correct NUMA ID's resources and always subtract them from `nodes` in-place.
+func subtractResourcesFromNUMANodeList(lh logr.Logger, nodes NUMANodeList, numaID int, qos corev1.PodQOSClass, containerRes corev1.ResourceList) error {
+	logEntries := []any{"numaCell", numaID}
+
+	for i := 0; i < len(nodes); i++ {
+		if nodes[i].NUMAID != numaID {
+			continue
+		}
+
+		nodeRes := nodes[i].Resources
+		lh.V(5).Info("NUMA resources before", append(logEntries, stringify.ResourceListToLoggable(nodeRes)...)...)
+
+		for resName, resQty := range containerRes {
+			isAffine := isNUMAAffineResource(resName)
+			if qos != corev1.PodQOSGuaranteed && isAffine {
+				lh.V(4).Info("ignoring QoS-depending exclusive request", "resource", resName, "QoS", qos)
+				continue
+			}
+			if resQty.IsZero() {
+				lh.V(4).Info("ignoring zero-valued request", "resource", resName)
+				continue
+			}
+			nResQ, ok := nodeRes[resName]
+			if !ok {
+				if isAffine {
```

Comment:
what if `cpu` or `memory` is not in NRT? One may configure nfd not to advertise it(like me for example) :)

Reply:
good point, need to think about this scenario

Reply:
I think we can downgrade this error to warning (log).